### PR TITLE
NavigationView: fix space reservation for top and bottom footer

### PR DESCRIPTION
**Pull request by beervoley**
## Description

This PR makes Navigation View free the space for top and bottom footer if it's not being used.
## Motivation and Context

Previously, Navigation View would reserve space for top and bottom footer even if they're not being used. It also will draw Separator line between main items and bottom footer even when footer is not present.
This PR addresses that.

NOTE: This PR only fixes the space reservation and not properly adjusts all the margins/paddings to correctly display all elements. It will be addressed in another PR.
## How Has This Been Tested?

Manually.
## Screenshots (if appropriate):

With toggle, back, autosuggestbox and settings on:
image

Without:
image

### CODE REVIEW

File: dev/NavigationView/NavigationView.cpp [hunk 1]
```diff
@@ -1526,31 +1527,58 @@ void NavigationView::UpdatePaneLayout()
...
+                                        separator.Visibility(winrt::Visibility::Visible);
+                                        separatorLineActualHeight = separatorLineHeight;
+                                    }
+                                    else
+                                    {
+                                        separator.Visibility(winrt::Visibility::Collapsed);
+                                    }
                                 }
-                                return totalAvailableHeightHalf;
+                                return totalAvailableHeightHalf - separatorLineActualHeight;
```
Comment (beervoley) [inline dev/NavigationView/NavigationView.cpp:1581]: @StephenLPeters @ranjeshj @karenbtlai this part with setting visibility of separator as well as actual height repeats 3 times - should I extract it somewhere down below?

File: dev/NavigationView/NavigationView.cpp [hunk 2]
```diff
@@ -1512,6 +1512,7 @@ void NavigationView::UpdatePaneLayout()
                         {
                             const auto footersActualHeight = footerItemsRepeater.ActualHeight();
                             const auto menuItemsActualHeight = menuItems.ActualHeight();
+                            const auto separatorLineHeight = GetItemSeparatorHeight();
```
Comment (beervoley) [inline dev/NavigationView/NavigationView.cpp:1515]: I'm not totally sure this is the best way to get separator line height. @karenbtlai thoughts?:)

**Comment by chingucoding** [comment]
Can I close #4193 then since it aims at doing the same?

**Comment by beervoley** [comment]
@chingucoding I didn't know you've already done it.. We can go with your solution if you can also remove space reservation for top footer when toggle or back button is not present?:)

**Comment by chingucoding** [comment]
@beervoley Can you share repro code, I'm not sure what you are referring too.

**Comment by beervoley** [comment]
@chingucoding it's just a simple NavigationView without back or toggle button:
image
this space on top shouldn't be reserved
check my changes in this PR, it's about removing minHeight again :)

File: dev/NavigationView/NavigationView.xaml [hunk 3]
```diff
@@ -541,6 +549,7 @@
                                             <!-- FooterItems -->
                                             <local:ItemsRepeaterScrollHost Grid.Row="3">
                                                 <ScrollViewer x:Name="FooterItemsScrollViewer"
+                                                        VerticalScrollBarVisibility="Auto"
```
Comment (beervoley) [inline dev/NavigationView/NavigationView.xaml:552]: If not set to atuo, verticalScrollBar will always be visible and will take space even when there are no items on the footer

Comment (chingucoding) [reply dev/NavigationView/NavigationView.xaml:552]: There is already a PR for this too: #3777

**Comment by chingucoding** [comment]
> @chingucoding I didn't know you've already done it.. We can go with your solution if you can also remove space reservation for top footer when toggle or back button is not present (look at my changes) ?:)

If we close this PR, I would copy those changes into my PR of course.

**Comment by beervoley** [comment]
converted to draft because @chingucoding has PR open with tests already. It only needs couple of small fixes.

File: dev/NavigationView/NavigationView.xaml (hunk 3 above)
Comment (beervoley) [reply dev/NavigationView/NavigationView.xaml:552]: @chingucoding could you please add the scrollbar change to your PR? I will close #3777

**Comment by beervoley** [comment]
> > @chingucoding I didn't know you've already done it.. We can go with your solution if you can also remove space reservation for top footer when toggle or back button is not present (look at my changes) ?:)
> 
> If we close this PR, I would copy those changes into my PR of course.

yes, that would be great. Once you add the changes and we get the pipelines pass, I will close this PR :)

File: dev/NavigationView/NavigationView.xaml (hunk 3 above)
Comment (chingucoding) [reply dev/NavigationView/NavigationView.xaml:552]: Added the changes from this PR and #3777 now.

Comment (beervoley) [reply dev/NavigationView/NavigationView.xaml:552]: awesome. I will verify your changes in a bit :)

**Comment by beervoley** [comment]
Closed because this bug will be addressed in https://github.com/microsoft/microsoft-ui-xaml/pull/4193

**Pull request closed without merging**